fe1-web/__tests__/model/objects/WalletCryptographyHandler.test.ts

```diff
@@ -0,0 +1,89 @@
+import { sign } from 'tweetnacl';
+import STRINGS from 'res/strings';
+import { get } from 'idb-keyval';
+
+/* used to simulate indexedDB database to test store/retrieve functions */
+require('fake-indexeddb/auto');
+
+/* mock crypto library object */
+const crypto = {};
+
+/**
+ * This test uses a MOCK version of the crypto.subtle API.
+ * The scope is not testing the SubtleCrypto library but testing the function
+ * calls to it. It also tests the 'store' and 'retrieve' functions of indexedDB
+ * with an intermediate simulation of app restart.
+ */
+describe('=== Wallet Cryptography Handler tests ===', () => {
+  describe('encryption - decryption', () => {
+    it('should correctly encrypt and decrypt the generated seed', async () => {
+      /* this is the MOCK crypto manager, it uses the mock SubtleCrypto library */
+      // eslint-disable-next-line max-len
+      let mockCryptoManager : WalletCryptographyHandler | null = new WalletCryptographyHandler(crypto);
+
+      /* seed to encrypt */
+      const seed = sign.keyPair().secretKey;
+
+      /* adding to crypto used by mockCryptoManager the subtle property
+         with the MOCKED functions: generateKey, encrypt, decrypt */
+      Object.defineProperty(crypto, 'subtle', {
+        value: {
+          generateKey: () => ({
+            publicKey: sign.keyPair().secretKey,
+            privateKey: sign.keyPair().publicKey,
+          }),
+
+          /* this mock encrypt function appends public encryption key to seed */
+          encrypt: async (s: AesGcmParams, publicKey: CryptoKey, plaintext: Uint8Array) => {
+            const ciphertext: string = plaintext.join().concat(publicKey.toString());
+            const ciphertextBuffer: ArrayBuffer = Buffer.from(ciphertext);
+            return new Promise((resolve) => {
+              resolve(ciphertextBuffer);
+            });
+          },
+
+          /* the mock decrypt function is called AFTER walletCryptoManager destruction.
+             The check is done on key coherence in database: since the encryption appended
+             the public key to the seed we need to retrieve the public key and undo the append
+             in order to provide correct decryption */
+          decrypt: async (s: RsaOaepParams, privateKey: CryptoKey, ciphertext: ArrayBuffer) => {
+            /* note the publicKey has to be retrieved here since the one passed in
+            parameter by the decrypt call in handler is the private key */
+            const publicKey = await get(STRINGS.walletPublicKeyId);
+            /* undoing public key append by substituting the substring with empty string */
+            const plaintext: string = ciphertext.toString().replace(publicKey.toString(), '');
+            const plaintextBuffer: ArrayBuffer = Buffer.from(plaintext);
+            return new Promise((resolve) => {
+              resolve(plaintextBuffer);
+            });
+          },
+        },
+      });
```

Comment:
you might want to move this in the definition of the `crypto` object above, as this is independent on the test itself (unless I'm mistaken).